### PR TITLE
Cb user chooses hvc

### PR DESCRIPTION
**Pull request by Cristian-Baeza**
- User can choose between hvc & hvh
- Computer can take own turns and works!
- Tried to not repeat myself in GameLoop logic, but it looks really messy to me still. Lmk what you think.

### CODE REVIEW

File: spec/game_logic_spec.rb [hunk 1]
```diff
@@ -129,4 +129,22 @@
       end
     end
 
+    describe "#cpu_turn" do
+      it "fills in board in a random open space" do
```
Comment (td-extreme) [inline spec/game_logic_spec.rb:133]:
```suggestion
      it "takes the last space availabe" do
```

Comment (Cristian-Baeza) [reply spec/game_logic_spec.rb:133]: 👍🏽

File: lib/main_app.rb [hunk 2]
```diff
@@ -9,16 +9,22 @@ def run
...
       puts(@console_output.show_board(@game))
+      
+      valid_options = @game.open_spaces.map(&:to_s)
+      
+      game_type == 1 ? user_choice = @questions.validate_user_input("PICK A SPACE #{@game.current_player()}", valid_options) : user_choice = @questions.validate_user_input("PICK A SPACE PLAYER X", valid_options) 
+      
+      @game.take_turn(user_choice)
 
-      valid_options = @game.open_spaces.map(&:to_s)  
+      game_type == 2 ? @game.cpu_turn : nil
```
[the remark this answers is not in the document]
Comment (Cristian-Baeza) [reply lib/main_app.rb:25]: 👍🏽

File: lib/main_app.rb [hunk 3]
```diff
@@ -9,16 +9,22 @@ def run
...
+    game_type = @questions.validate_user_input("PLAY VS: \n1.HUMAN \n2.CPU", ["1","2"])
+
     until @game.is_game_over? do
-      puts(ConsoleOutput::WELCOME_MESSAGE)
+      game_type == 1 ? puts(ConsoleOutput::WELCOME_MESSAGE_HVH) : puts(ConsoleOutput::WELCOME_MESSAGE_HVC)
+      
       puts(@console_output.show_board(@game))
+      
+      valid_options = @game.open_spaces.map(&:to_s)
+
```
[the remark this answers is not in the document]
Comment (Cristian-Baeza) [reply lib/main_app.rb:20]: 👍🏽

File: lib/main_app.rb [hunk 4]
```diff
@@ -9,22 +9,28 @@ def run
...
     @console_output = ConsoleOutput.new
 
-    game_type = @questions.validate_user_input("PLAY VS: \n1.HUMAN \n2.CPU", ["1","2"])
+    game_type = @questions.validate_user_input("PLAY VS: \n1.HUMAN \n2.CPU", ["1","2"]) == 1 ?
+      :human_vs_human : :human_vs_computer
 
     until @game.is_game_over? do
-      game_type == 1 ? puts(ConsoleOutput::WELCOME_MESSAGE_HVH) : puts(ConsoleOutput::WELCOME_MESSAGE_HVC)
+      game_type == :human_vs_human ? 
+        puts(@console_output.welcome_message("PLAYER")) :
```
[the remark this answers is not in the document]
Comment (td-extreme) [reply lib/main_app.rb:17]: It is.
But, now that I'm looking at it. I have another thought.

**Pull request merged**